libraries/AP_Camera/examples/tracking.py

```diff
@@ -0,0 +1,131 @@
+"""
+Required Installations:
+pip install opencv-python-headless pymavlink ultralytics
+sudo apt-get install libgstreamer1.0-0 gstreamer1.0-tools gstreamer1.0-plugins-base gstreamer1.0-plugins-good
+"""
+
+import cv2
+import threading
+from pymavlink import mavutil
+import math 
+from ultralytics import YOLO
+import argparse
+
+class DroneGimbalControl:
+    def __init__(self, video_source=None):
+        # Set default video source if not provided
+        if video_source is None:
+            video_source = 'udpsrc port=5600 ! application/x-rtp,encoding-name=H264,payload=96 ! rtph264depay ! h264parse ! queue ! avdec_h264 ! videoconvert ! appsink'
+
+        self.master = mavutil.mavlink_connection('127.0.0.1:14560')
+        self.master.wait_heartbeat()
+        print("Heartbeat from the system (system %u component %u)" % 
+             (self.master.target_system, self.master.target_component))
+
+        # Store relevant parameters as object state
+        self.target_system = self.master.target_system
+        self.target_component = self.master.target_component
+        self.flags = 0
+        self.gimbal_device_id = 0
+
+        self.cap = cv2.VideoCapture(video_source, cv2.CAP_GSTREAMER)
+        self.model = YOLO("yolov8n.pt")
+        self.frame_lock = threading.Lock()
+        self.latest_frame = None
+        self.center_y = 0
+        self.center_x = 0
+
+    def send_gimbal_manager_pitch_yaw_angles(self, pitch, yaw, pitch_rate, yaw_rate):
+        msg = self.master.mav.gimbal_manager_set_pitchyaw_encode(
+            self.target_system,
+            self.target_component,
+            self.flags,
+            self.gimbal_device_id,
+            pitch,
+            yaw,
+            pitch_rate,
+            yaw_rate
+        )
+        self.master.mav.send(msg)
+
+    def send_command(self):
+        while True:
+            centre_x_copy = int(self.center_x)
+            centre_y_copy = int(self.center_y)
+            if (centre_x_copy == 0 and centre_y_copy == 0):
+                diff_x = 0
+                diff_y = 0
+            else:
+                diff_x = (centre_x_copy - (640 / 2)) / 2
+                diff_y = -(centre_y_copy - (480 / 2)) / 2
+
+            self.send_gimbal_manager_pitch_yaw_angles(float("NaN"), float("NaN"), math.radians(diff_y), math.radians(diff_x))
```

Comment:
As PeterB mentioned, we should ensure that this runs no faster than 50hz or maybe even 20hz